adapt/src/main/java/io/github/vshnv/adapt/adapter/LifecycleAwareAdaptAdapter.kt

```diff
@@ -28,12 +29,22 @@ class LifecycleAwareAdaptAdapter<T : Any>(private val viewTypeMapper: ((T, Int)
     private val mDiffer: AsyncListDiffer<T> = AsyncListDiffer(this, diffCallback)
     override val currentList: List<T>
         get() = mDiffer.currentList
+    private var fullList: MutableList<T> = mutableListOf()
```

Comment:
Storing the entire list and passing the actual list to the differ might not work in this case. submitData would reset fulldata so theres no way to notify the adapter of the filtered list. Might be worth it to extend the class and provide a filtering option there with explicit note of "unfilteredData" and "data".

Reply:
I didn't get what actually you are talking about!  🥲 ..